### ISSUE TITLE
Icon not shown

### ISSUE THREAD

**Issue by goldmund256** [dunst-project/dunst#977]
After the new version 1.7.1 came out, custom icons disappeared in dunst.

When launching a notification using:
`dunstify -i ~/path/to/icon "text"`
or for a raw image:
`dunstify -I ~/path/to/icon "text"`

the standard icon set up in dunstrc (section `[urgency_normal]`) is shown in its place. The expected behaviour is that the standard icon be overwritten when specifying an image using the dunstify command.

Unfortunately, I am not getting any errors or warnings:

[This is the output of](https://bin.0xfc.de/?be6442779dd85eee#AVuiVcN8KhnrkYfqAvR8qpyEbBarMgjD9D1rGP8CM2jd) `dunst -verbosity debug` before, while and after `dunstify -i ~/path/to/icon "text"` is triggered.

[This is the output of](https://bin.0xfc.de/?9901ce469c04efbe#AJGekYfVGYnQN2tGY88avvn6uz8eNZdYrhDE9bZzPhnR) `dbus-monitor path=/org/freedesktop/Notifications` using the same dunstify command.

Is anyone experiencing the same issue?

### Installation info

- Version: `1.7.1 (2021-11-01)`
- Install type: `package`
- Distro and version: `Arch Linux 5.15.2-arch1-1`

**Comment by Ahanaf-Ether** [comment]
Me too

**Comment by fwsmit** [comment]
When sending only the name of the icon, does it work then? Of course the icon has to be in your path/theme

**Comment by Ahanaf-Ether** [comment]
It is

**Comment by Ahanaf-Ether** [comment]
Also it works when i downgrade dunst

**Comment by goldmund256** [comment]
> When sending only the name of the icon, does it work then? Of course the icon has to be in your path/theme

No, that doesn't work for me. The desired icon is in my `icon_path`, however, the command:

`dunstify -i "locked.png" "test"`
or
`dunstify -i "locked" "test"`

still shows the generic icon.

Strange that Ahanaf-Ether had a different experience. Could somebody else test this?

**Comment by akinokonomi** [comment]
> When sending only the name of the icon, does it work then? Of course the icon has to be in your path/theme

Does not work for me, either. I use `notify-send`.

**Comment by goldmund256** [comment]
> Does not work for me, either. I use `notify-send`.

`notify-send` doesn't have that functionality, you can't select a custom icon. Try it with `dunstify -i "icon.png" "text to display"`

**Comment by akinokonomi** [comment]
> `notify-send` doesn't have that functionality, you can't select a custom icon.

What do you mean?

```bash
$ notify-send --help | grep icon
  -a, --app-name=APP_NAME           Specifies the app name for the icon
  -i, --icon=ICON[,ICON...]         Specifies an icon filename or stock icon to display.
```

**Comment by goldmund256** [comment]
> What do you mean?
> 
> ```shell
> $ notify-send --help | grep -- -i       
>   -i, --icon=ICON[,ICON...]         Specifies an icon filename or stock icon to display.
> ```

Sorry, my bad. It's the first time I see this option. Anyway, thanks for testing.

**Comment by fwsmit** [comment]
If anyone had time they can try to `git bisect` this  issue, as it seems to be a regression in v1.7.0

**Comment by Ahanaf-Ether** [comment]
After a lot of testing I figured it's a version specific issue, as older versions seem to work specially 1.6.1 and I don't understand the reason for depreciating the   "Shortcut section" in the newer release, is it for making it more lightweight but it doesn't make sense to do "dusntctl" all the time

**Comment by fwsmit** [comment]
> After a lot of testing I figured it's a version specific issue, as older versions seem to work specially 1.6.1

Yes, that is what it seem like. If you can find the specific commit this happens in, that would make it easier for me to fix. As I said, you can use `git bisect` for that. Look it up if you don't know how to use that.

> and I don't understand the reason for depreciating the "Shortcut section" in the newer release, is it for making it more lightweight but it doesn't make sense to do "dusntctl" all the time

This doesn't have much to do with this issue, but you can look at https://github.com/dunst-project/dunst/issues/978 for how to check if a notification is visible.

**Comment by Ahanaf-Ether** [comment]
Ok, I'll check it out.

**Comment by fwsmit** [comment]
I haven't been able to reproduce this issue. Could you try reproducing it with the default dunstrc?
Note that `dunstify -i "icon.png" "text to display"` doesn't work, because dunstify doesn't look at relative paths. You need to specify the full path to the icon.

**Comment by Ahanaf-Ether** [comment]
I did

**Comment by Ahanaf-Ether** [comment]
Would you like to take a look at my duns
trc?

**Comment by Ahanaf-Ether** [comment]
> I haven't been able to reproduce this issue. Could you try reproducing it with the default dunstrc? Note that `dunstify -i "icon.png" "text to display"` doesn't work, because dunstify doesn't look at relative paths. You need to specify the full path to the icon or only the name of the icon, e.g. `icon`.

I did the default dunstrc doesn't do any good either, however like I said 1.6.1 seems to do fine

**Comment by fwsmit** [comment]
Then I need better reproduction steps, as I wasn't able to reproduce it with a pretty basic dunstrc.

**Comment by Ahanaf-Ether** [comment]
Hang on I will send you mine

**Comment by Ahanaf-Ether** [comment]
[global]
    ### Display ###

    # Which monitor should the notifications be displayed on.
    monitor = 0
    
    # Display notification on focused monitor.  Possible modes are:
    #   mouse: follow mouse pointer
    #   keyboard: follow window with keyboard focus
    #   none: don't follow anything
    #
    # "keyboard" needs a window manager that exports the
    # _NET_ACTIVE_WINDOW property.
    # This should be the case for almost all modern window managers.
    #
    # If this option is set to mouse or keyboard, the monitor option
    # will be ignored.
    follow = none

    ### Geometry ###

    # dynamic width from 0 to 300
    # width = (0, 300)
    # constant width of 300
    width = 300

    # The maximum height of a single notification, excluding the frame.
    height = 300
  
    # Position the notification in the top right corner
    origin = top-right

    # Offset from the origin
    offset = 10x50

    # Scale factor. It is auto-detected if value is 0.
    scale = 0

    # Maximum number of notification (0 means no limit)
    notification_limit = 0    # Show how many messages are currently hidden (because of geometry).
    
    # Shrink window if it's smaller than the width.  Will be ignored if
    # width is 0.
    shrink = no

    # The transparency of the window.  Range: [0; 100].
    # This option will only work if a compositing window manager is
    # present (e.g. xcompmgr, compiz, etc.).
    transparency = 16

    # Draw a line of "separator_height" pixel height between two
    # notifications.
    # Set to 0 to disable.
    separator_height = 2

    # Padding between text and separator.
    padding = 8
    separator_color = "#32302f"

    # Horizontal padding.
    horizontal_padding = 8

    # Defines width in pixels of frame around the notification window.
    # Set to 0 to disable.
    frame_width = 1
 
    ### Progress bar ###

    # Turn on the progess bar. It appears when a progress hint is passed with
    # for example dunstify -h int:value:12
    progress_bar = true

    # Set the progress bar height. This includes the frame, so make sure
    # it's at least twice as big as the frame width.
    progress_bar_height = 10

    # Set the frame width of the progress bar
    progress_bar_frame_width = 1

    # Set the minimum width for the progress bar
    progress_bar_min_width = 150

    # Set the maximum width for the progress bar
    progress_bar_max_width = 300


    # Show how many messages are currently hidden (because of
    # notification_limit).
    indicate_hidden = yes   

    ### Text ###

    font = Noto Sans Regular 11

    # The spacing between lines.  If the height is smaller than the
    # font height, it will get raised to the font height.
    line_height = 0

    # Possible values are:
    # full: Allow a small subset of html markup in notifications:
    #        <b>bold</b>
    #        <i>italic</i>
    #        <s>strikethrough</s>
    #        <u>underline</u>
    #
    #        For a complete reference see
    #        <http://developer.gnome.org/pango/stable/PangoMarkupFormat.html>.
    #
    # strip: This setting is provided for compatibility with some broken
    #        clients that send markup even though it's not enabled on the
    #        server. Dunst will try to strip the markup but the parsing is
    #        simplistic so using this option outside of matching rules for
    #        specific applications *IS GREATLY DISCOURAGED*.
    #
    # no:    Disable markup parsing, incoming notifications will be treated as
    #        plain text. Dunst will not advertise that it has the body-markup
    #        capability if this is set as a global setting.
    #
    # It's important to note that markup inside the format option will be parsed
    # regardless of what this is set to.
    markup = yes

    # The format of the message.  Possible variables are:
    #   %a  appname
    #   %s  summary
    #   %b  body
    #   %i  iconname (including its path)
    #   %I  iconname (without its path)
    #   %p  progress value if set ([  0%] to [100%]) or nothing
    #   %n  progress value if set without any extra characters
    #   %%  Literal %
    # Markup is allowed
    format = "<b>%s</b>\n%b"

    # Alignment of message text.
    # Possible values are "left", "center" and "right".
    alignment = left

    # Show age of message if message is older than show_age_threshold
    # seconds.
    # Set to -1 to disable.
    show_age_threshold = 60

    # Split notifications into multiple lines if they don't fit into
    # geometry.
    word_wrap = yes

    # When word_wrap is set to no, specify where to make an ellipsis in long lines.
    # Possible values are "start", "middle" and "end".
    ellipsize = middle

    # Ignore newlines '\n' in notifications.
    ignore_newline = no

    # Stack together notifications with the same content
    stack_duplicates = true

    # Hide the count of stacked notifications with the same content
    hide_duplicate_count = false

    # Display indicators for URLs (U) and actions (A).
    show_indicators = yes

    ### Icons ###

    # Align icons left/right/off
    icon_position = left

    # Scale larger icons down to this size, set to 0 to disable
   min_icon_size=32 
   max_icon_size = 32

    # Paths to default icons.
    #icon_path = /usr/share/icons/gnome/16x16/status/:/usr/share/icons/gnome/16x16/devices/
    icon_path =
    /usr/share/icons/Gruvbox-material-dark/16x16/status/:/usr/share/icons/Gruvbox-Material-Dark/16x16/devices/:/usr/share/icons/Gruvbox-Material-Dark/16x16/legacy/
    ### History ###

    # Should a notification popped up from history be sticky or timeout
    # as if it would normally do.
    sticky_history = yes

    # Maximum amount of notifications kept in history
    history_length = 20

    ### Misc/Advanced ###

    # dmenu path.
    dmenu = /usr/bin/dmenu -p dunst:

    # Browser for opening urls in context menu.
    browser = /usr/bin/google-chrome-stable -new-tab

    # Always run rule-defined scripts, even if the notification is suppressed
    always_run_script = true

    # Define the title of the windows spawned by dunst
    title = Dunst

    # Define the class of the windows spawned by dunst
    class = Dunst

    # Define the corner radius of the notification window
    # in pixel size. If the radius is 0, you have no rounded
    # corners.
    # The radius will be automatically lowered if it exceeds half of the
    # notification height to avoid clipping text and/or icons.
    corner_radius = 0

    ### Legacy

    # Use the Xinerama extension instead of RandR for multi-monitor support.
    # This setting is provided for compatibility with older nVidia drivers that
    # do not support RandR and using it on systems that support RandR is highly
    # discouraged.
    #
    # By enabling this setting dunst will not be able to detect when a monitor
    # is connected or disconnected which might break follow mode if the screen
    # layout changes.
    force_xinerama = false

    frame_color = "#b8bb26"
    frame_width = 3

 ### mouse

    # Defines list of actions for each mouse event
    # Possible values are:
    # * none: Don't do anything.
    # * do_action: Invoke the action determined by the action_name rule. If there is no
    #              such action, open the context menu.
    # * open_url: If the notification has exactly one url, open it. If there are multiple
    #             ones, open the context menu.
    # * close_current: Close current notification.
    # * close_all: Close all notifications.
    # * context: Open context menu for the notification.
    # * context_all: Open context menu for all notifications.
    # These values can be strung together for each mouse event, and
    # will be executed in sequence.
    mouse_left_click = close_current
    mouse_middle_click = do_action, close_current
    mouse_right_click = close_all
   
[urgency_critical]
  
    timeout = 0
    background = "#1d2021"
    foreground = "#ebdbb2"

# Experimental features that may or may not work correctly. Do not expect them
# to have a consistent behaviour across releases.
[experimental]
    # Calculate the dpi to use on a per-monitor basis.
    # If this setting is enabled the Xft.dpi value will be ignored and instead
    # dunst will attempt to calculate an appropriate dpi value for each monitor
    # using the resolution and physical size. This might be useful in setups
    # where there are multiple screens with very different dpi values.
    per_monitor_dpi = false

#[shortcuts]

    # Shortcuts are specified as [modifier+][modifier+]...key
    # Available modifiers are "ctrl", "mod1" (the alt-key), "mod2",
    # "mod3" and "mod4" (windows-key).
    # Xev might be helpful to find names for keys.

    # Close notification.
    # close = ctrl+space

    # Close all notifications.
    # close_all = ctrl+shift+space

    # Redisplay last message(s).
    # On the US keyboard layout "grave" is normally above TAB and left
    # of "1". Make sure this key actually exists on your keyboard layout,
    # e.g. check output of 'xmodmap -pke'
    # history = ctrl+grave
 
[urgency_low]
    background = "#1d2021"
    foreground = "#ebdbb2"
    timeout = 5

[urgency_normal]
    background = "#1d2021"
    foreground = "#ebdbb2"
    timeout = 20

**Comment by Ahanaf-Ether** [comment]
`[global]
    ### Display ###

    # Which monitor should the notifications be displayed on.
    monitor = 0
    
    # Display notification on focused monitor.  Possible modes are:
    #   mouse: follow mouse pointer
    #   keyboard: follow window with keyboard focus
    #   none: don't follow anything
    #
    # "keyboard" needs a window manager that exports the
    # _NET_ACTIVE_WINDOW property.
    # This should be the case for almost all modern window managers.
    #
    # If this option is set to mouse or keyboard, the monitor option
    # will be ignored.
    follow = none

    ### Geometry ###

    # dynamic width from 0 to 300
    # width = (0, 300)
    # constant width of 300
    width = 300

    # The maximum height of a single notification, excluding the frame.
    height = 300
  
    # Position the notification in the top right corner
    origin = top-right

    # Offset from the origin
    offset = 10x50

    # Scale factor. It is auto-detected if value is 0.
    scale = 0

    # Maximum number of notification (0 means no limit)
    notification_limit = 0    # Show how many messages are currently hidden (because of geometry).
    
    # Shrink window if it's smaller than the width.  Will be ignored if
    # width is 0.
    shrink = no

    # The transparency of the window.  Range: [0; 100].
    # This option will only work if a compositing window manager is
    # present (e.g. xcompmgr, compiz, etc.).
    transparency = 16

    # Draw a line of "separator_height" pixel height between two
    # notifications.
    # Set to 0 to disable.
    separator_height = 2

    # Padding between text and separator.
    padding = 8
    separator_color = "#32302f"

    # Horizontal padding.
    horizontal_padding = 8

    # Defines width in pixels of frame around the notification window.
    # Set to 0 to disable.
    frame_width = 1
 
    ### Progress bar ###

    # Turn on the progess bar. It appears when a progress hint is passed with
    # for example dunstify -h int:value:12
    progress_bar = true

    # Set the progress bar height. This includes the frame, so make sure
    # it's at least twice as big as the frame width.
    progress_bar_height = 10

    # Set the frame width of the progress bar
    progress_bar_frame_width = 1

    # Set the minimum width for the progress bar
    progress_bar_min_width = 150

    # Set the maximum width for the progress bar
    progress_bar_max_width = 300


    # Show how many messages are currently hidden (because of
    # notification_limit).
    indicate_hidden = yes   

    ### Text ###

    font = Noto Sans Regular 11

    # The spacing between lines.  If the height is smaller than the
    # font height, it will get raised to the font height.
    line_height = 0

    # Possible values are:
    # full: Allow a small subset of html markup in notifications:
    #        <b>bold</b>
    #        <i>italic</i>
    #        <s>strikethrough</s>
    #        <u>underline</u>
    #
    #        For a complete reference see
    #        <http://developer.gnome.org/pango/stable/PangoMarkupFormat.html>.
    #
    # strip: This setting is provided for compatibility with some broken
    #        clients that send markup even though it's not enabled on the
    #        server. Dunst will try to strip the markup but the parsing is
    #        simplistic so using this option outside of matching rules for
    #        specific applications *IS GREATLY DISCOURAGED*.
    #
    # no:    Disable markup parsing, incoming notifications will be treated as
    #        plain text. Dunst will not advertise that it has the body-markup
    #        capability if this is set as a global setting.
    #
    # It's important to note that markup inside the format option will be parsed
    # regardless of what this is set to.
    markup = yes

    # The format of the message.  Possible variables are:
    #   %a  appname
    #   %s  summary
    #   %b  body
    #   %i  iconname (including its path)
    #   %I  iconname (without its path)
    #   %p  progress value if set ([  0%] to [100%]) or nothing
    #   %n  progress value if set without any extra characters
    #   %%  Literal %
    # Markup is allowed
    format = "<b>%s</b>\n%b"

    # Alignment of message text.
    # Possible values are "left", "center" and "right".
    alignment = left

    # Show age of message if message is older than show_age_threshold
    # seconds.
    # Set to -1 to disable.
    show_age_threshold = 60

    # Split notifications into multiple lines if they don't fit into
    # geometry.
    word_wrap = yes

    # When word_wrap is set to no, specify where to make an ellipsis in long lines.
    # Possible values are "start", "middle" and "end".
    ellipsize = middle

    # Ignore newlines '\n' in notifications.
    ignore_newline = no

    # Stack together notifications with the same content
    stack_duplicates = true

    # Hide the count of stacked notifications with the same content
    hide_duplicate_count = false

    # Display indicators for URLs (U) and actions (A).
    show_indicators = yes

    ### Icons ###

    # Align icons left/right/off
    icon_position = left

    # Scale larger icons down to this size, set to 0 to disable
   min_icon_size=32 
   max_icon_size = 32

    # Paths to default icons.
    #icon_path = /usr/share/icons/gnome/16x16/status/:/usr/share/icons/gnome/16x16/devices/
    icon_path =
    /usr/share/icons/Gruvbox-material-dark/16x16/status/:/usr/share/icons/Gruvbox-Material-Dark/16x16/devices/:/usr/share/icons/Gruvbox-Material-Dark/16x16/legacy/
    ### History ###

    # Should a notification popped up from history be sticky or timeout
    # as if it would normally do.
    sticky_history = yes

    # Maximum amount of notifications kept in history
    history_length = 20

    ### Misc/Advanced ###

    # dmenu path.
    dmenu = /usr/bin/dmenu -p dunst:

    # Browser for opening urls in context menu.
    browser = /usr/bin/google-chrome-stable -new-tab

    # Always run rule-defined scripts, even if the notification is suppressed
    always_run_script = true

    # Define the title of the windows spawned by dunst
    title = Dunst

    # Define the class of the windows spawned by dunst
    class = Dunst

    # Define the corner radius of the notification window
    # in pixel size. If the radius is 0, you have no rounded
    # corners.
    # The radius will be automatically lowered if it exceeds half of the
    # notification height to avoid clipping text and/or icons.
    corner_radius = 0

    ### Legacy

    # Use the Xinerama extension instead of RandR for multi-monitor support.
    # This setting is provided for compatibility with older nVidia drivers that
    # do not support RandR and using it on systems that support RandR is highly
    # discouraged.
    #
    # By enabling this setting dunst will not be able to detect when a monitor
    # is connected or disconnected which might break follow mode if the screen
    # layout changes.
    force_xinerama = false

    frame_color = "#b8bb26"
    frame_width = 3

 ### mouse

    # Defines list of actions for each mouse event
    # Possible values are:
    # * none: Don't do anything.
    # * do_action: Invoke the action determined by the action_name rule. If there is no
    #              such action, open the context menu.
    # * open_url: If the notification has exactly one url, open it. If there are multiple
    #             ones, open the context menu.
    # * close_current: Close current notification.
    # * close_all: Close all notifications.
    # * context: Open context menu for the notification.
    # * context_all: Open context menu for all notifications.
    # These values can be strung together for each mouse event, and
    # will be executed in sequence.
    mouse_left_click = close_current
    mouse_middle_click = do_action, close_current
    mouse_right_click = close_all
   
[urgency_critical]
  
    timeout = 0
    background = "#1d2021"
    foreground = "#ebdbb2"

# Experimental features that may or may not work correctly. Do not expect them
# to have a consistent behaviour across releases.
[experimental]
    # Calculate the dpi to use on a per-monitor basis.
    # If this setting is enabled the Xft.dpi value will be ignored and instead
    # dunst will attempt to calculate an appropriate dpi value for each monitor
    # using the resolution and physical size. This might be useful in setups
    # where there are multiple screens with very different dpi values.
    per_monitor_dpi = false

#[shortcuts]

    # Shortcuts are specified as [modifier+][modifier+]...key
    # Available modifiers are "ctrl", "mod1" (the alt-key), "mod2",
    # "mod3" and "mod4" (windows-key).
    # Xev might be helpful to find names for keys.

    # Close notification.
    # close = ctrl+space

    # Close all notifications.
    # close_all = ctrl+shift+space

    # Redisplay last message(s).
    # On the US keyboard layout "grave" is normally above TAB and left
    # of "1". Make sure this key actually exists on your keyboard layout,
    # e.g. check output of 'xmodmap -pke'
    # history = ctrl+grave
 
[urgency_low]
    background = "#1d2021"
    foreground = "#ebdbb2"
    timeout = 5

[urgency_normal]
    background = "#1d2021"
    foreground = "#ebdbb2"
    timeout = 20
`

**Comment by goldmund256** [comment]
Sorry, I just got home. To help you understand and possibly reproduce the issue, [here is my `inxi -F`](https://bin.0xfc.de/?d0fb20c8330b0b97#u3fAoPw2vbyKSWzZ82Mu8pYNaUnwvx5S6BKNYpRP5fM).

[And here is my dunstrc.](https://bin.0xfc.de/?0289cc7313753bf0#Hs6JZ3DYZAhcTgqFdpXVcSyuW7Y3ejNnwTjETVrKPaiT)

As you see in my dunstrc, `icon_path` is equal to:
`icon_path = /home/user/.local/share/icons`

There are the following icons in this directory:
```
critical.png
door.png
locked.png
mount.png
normal.png
power.png
unlocked.png
```

And the problem is: neither of the following commands shows the desired icon, but instead displays the generic `normal.png`:

```
$ dunstify -i /home/user/.local/share/icons/locked.png "text"
$ dunstify -i locked "text"
```

**Important:**
I tested the default dunstrc and the problem disappeared – icons could be specified with `dunstify -i ...` again. However, I would like to have a generic icon for notifications which I don't specify an icon for (and this setup isn't in the original dunstrc). As soon as I define a filepath for default icons in dunstrc (using `[new_icon]`), they are displayed even when they should be overwritten by the custom ones.

Please let me know if you need any more info.

**Comment by fwsmit** [comment]
> [And here is my dunstrc.](https://bin.0xfc.de/?0289cc7313753bf0#Hs6JZ3DYZAhcTgqFdpXVcSyuW7Y3ejNnwTjETVrKPaiT)
> 
> As you see in my dunstrc, `icon_path` is equal to: `icon_path = /home/user/.local/share/icons`
> 
> There are the following icons in this directory:
> 
> ```
> critical.png
> door.png
> locked.png
> mount.png
> normal.png
> power.png
> unlocked.png
> ```
> 
> And the problem is: neither of the following commands shows the desired icon, but instead displays the generic `normal.png`:
> 
> ```
> $ dunstify -i /home/user/.local/share/icons/locked.png "text"
> $ dunstify -i locked "text"
> ```
> 
> **Important:** I tested the default dunstrc and the problem disappeared – icons could be specified with `dunstify -i ...` again. However, I would like to have a generic icon for notifications which I don't specify an icon for (and this setup isn't in the original dunstrc). As soon as I define a filepath for default icons in dunstrc (using `new_icon` under `[urgency_normal]`), they are displayed even when they should be overwritten by the custom ones.
> 
> Please let me know if you need any more info.

I found the issue. The `new_icon` overrides the icon passed via dunstify. Before you could use `icon` here, but that doesn't seem to work anymore, as it's interpreted as a filtering rule. I'll fix this issue, but I also want to rename this setting to `default_icon` to prevent confusion. This way it should also work outside the the urgency sections.
Temporary workaround: name your default icons `dialog-information` and `dialog-warning`.

**Comment by goldmund256** [comment]
> I found the issue. The `new_icon` overrides the icon passed via dunstify. Before you could use `icon` here, but that doesn't seem to work anymore, as it's interpreted as a filtering rule. I'll fix this issue, but I also want to rename this setting to `default_icon` to prevent confusion. This way it should also work outside the the urgency sections. Temporary workaround: name your default icons `dialog-information` and `dialog-warning`.

Perfect, that's great to hear!

Btw, your suggested temporary workaround doesn't seem to be working. I've renamed the default icons to `dialog-information.png` and `dialog-warning.png` and put them into dunstrc like so:

```
[urgency_normal]
background = "#2b2b2b"
foreground = "#ffffff"
timeout = 5
new_icon = dialog-information
```

and the default icon on `dunstify -i ...` is still showing. But this issue is nothing dramatic and I am happy to wait for a proper fix.

Just a note: when you rename `new_icon` to `default_icon`, please do update the template dunstrc and the documentation. Last time `icon` got changed to `new_icon`, I [had some trouble](https://github.com/dunst-project/dunst/issues/944) finding the correct syntax and transferring my old dunstrc to the new one.

Thanks a lot!

**Comment by fwsmit** [comment]
> ```
> [urgency_normal]
> background = "#2b2b2b"
> foreground = "#ffffff"
> timeout = 5
> new_icon = dialog-information
> ```

Part of the workaround is removing the `new_icon` from your dunstrc.


> and the default icon on `dunstify -i ...` is still showing. But this issue is nothing dramatic and I am happy to wait for a proper fix.

Working on it.

> Just a note: when you rename `new_icon` to `default_icon`, please do update the template dunstrc and the documentation. Last time `icon` got changed to `new_icon`, I [had some trouble](https://github.com/dunst-project/dunst/issues/944) finding the correct syntax and transferring my old dunstrc to the new one.

`default_icon` is not a replacement for `new_icon`. I'll make the manual a bit more clear as well:
```
=item C<new_icon>

Updates the icon of the notification, it should be a path or a name for a valid
image. This overrides the icon that was sent with dunstify or another notification
tool.

=item C<default_icon>

Sets the default icon of the notification, it should be a path or a name for a
valid image. This does B<not> override the icon that was sent with dunstify or
another notification tool.
```

**Comment by fwsmit** [comment]
I've fixed both issues in #984. The icon setting should now now work again. I've also added a default_icon setting as a future replacement.
The default dunstrc is now fixed as well. `new_icon` is replaced with `default_icon`.

**Comment by akinokonomi** [comment]
> I've fixed both issues in #984. 

Tested. Both `default_icon` and `icon` works great now. Thank you.

**Comment by goldmund256** [comment]
> I've fixed both issues in #984.

I've just tested it as well. `default_icon` and `icon` work beautifully. Thank you!

**Comment by fwsmit** [comment]
Cool, I'll close this issue now
